sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -88,4 +90,32 @@ object IntervalUtils {
     result += MICROS_PER_MONTH * (interval.months % MONTHS_PER_YEAR)
     Decimal(result, 18, 6)
   }
+
+  /**
+   * Gets interval duration
+   *
+   * @param cal - the interval to get duration
+   * @param daysPerMonth - the number of days per one month
+   * @param targetUnit - time units of the result
+   * @return duration in the specified time units
+   */
+  def getDuration(
+      cal: CalendarInterval,
+      daysPerMonth: Int,
```

Comment:
This could also be an optional final argument and default to 31. For now, when it's always 31, maybe that's simpler.